pennylane/_qubit_device.py

```diff
@@ -285,6 +285,20 @@ def statistics(self, observables):
             elif obs.return_type is Probability:
                 results.append(self.probability(wires=obs.wires))
 
+            elif obs.return_type is State:
+                if len(obs.wires) != self.num_wires:
+                    raise QuantumFunctionError("The state must be returned over all wires")
+
+                try:
+                    state = self.state
+                except AttributeError:
+                    state = None
+
+                if state is None:
+                    raise AttributeError("The state is not available in the current device")
+
+                results.append(self.state)
+
```

Comment:
https://github.com/PennyLaneAI/pennylane/pull/737/files#r467667482
^ Previous conversation

![image](https://user-images.githubusercontent.com/49409390/92802546-98063e00-f384-11ea-8e55-43beae720944.png)

I have talked with @mariaschuld about adding a `provides_state` capability. For now I'm sticking with the `try/except` approach. Also, yes, there is a chance that the state is available but has value `None`, so I thought this was one way to capture both the `AttributeError` and `state=None`.

Reply:
Good point! I'll see how the device capabilities PR is going.

Reply:
Device capability PR is merged :) 

Since devices do not have to define the capability (and we haven't updated the ecosystem yet, so most devices don't), you'd have three cases: 

- 'returns_state' not in capabilities -> unknown, could be interpreted as False if you like...the device tests should pick up if it is reported wrongly in a device
- 'returns_state' = True
- 'returns_state' = False

Reply:
Thanks a lot @mariaschuld!